examples/federated-learning/tff/nvflare/README.md

```diff
@@ -0,0 +1,283 @@
+is a domain-agnostic, open-source, extensible SDK that allows researchers and data scientists to adapt existing ML/DL workflows to a federated paradigm.
+It enables platform developers to build a secure, privacy-preserving offering for a distributed multi-party collaboration.
+
+[Here](https://nvflare.readthedocs.io/en/main/flare_overview.html#high-level-system-architecture) is a high level system architecture of Nvflare.
+
+> **_NOTE:_** This example uses the [MNIST](http://yann.lecun.com/exdb/mnist/) handwritten digits dataset and will load its data within the trainer code.
+
+This example builds on top of the infrastructure that the
+[blueprint provides](../../../../README.md), and follows the best practices the
+blueprint establishes.
+
+## High-level diagram
+This demo relies on one server and two clients that are connected to the server through a secure gRPC link.
+
+Here is a high-level diagram:
+
+![alt_text](../../../../assets/nvflare.svg "Infrastructure overview")
+
+## 1. Create custom image
+For this demo to work, you have to create a custom image with TensorFlow and Nvflare installed. To create the image and push it on Artifact Registry, go to the `docker-image` and build the image:
+
+```bash
+cd docker-image
+gcloud builds submit --tag europe-west1-docker.pkg.dev/lgu-demos/nvflare/nvflare-tensorflow:1.0.0
+```
+
+## 2. Create persistent volume for Nvflare workspace
+All the models generated will be stored in a Cloud storage bucket mounted by each pod. First, create a bucket:
+
+```bash
+gcloud storage buckets create gs://nvflare-storage
+```
+
+Add the following permissions to the Kubernetes Service Account:
+
+```bash
+gcloud storage buckets add-iam-policy-binding gs://nvflare-storage \
+    --member "principal://iam.googleapis.com/projects/391818703174/locations/global/workloadIdentityPools/lgu-demos.svc.id.goog/subject/ns/default/sa/default" \
+    --role "roles/storage.objectUser"
+```
+
+Create a persistent volume to store all the data needed fot the clients and server to work:
+
+```bash
+kubectl apply -f pv-workspace.yml
+kubectl apply -f pvc-workspace.yml
+```
+
+## 3. Create Nvflare helper
+Now that the persistent volume is created, you can start the helper pod that will create the basic infrastructure needed to run the demo.
+
+```bash
+kubectl apply -f helper-deployment.yml
+```
+
+When the helper pod is created, go inside the pod and create the needed basic infrastructure for Nvflare to run:
+
+```bash
+kubectl exec -it deploy/nvflare-helper -c nvflare-helper -- bash
+```
+
+Once inside the pod, run the provisionning tool:
+
+```bash
+cd /workspace/nvfl/
+nvflare provision
+```
+
+You have to choose the non-ha choice, as ha is not yet supported on Kubernetes. It will create a `project.yml` file that you can customize for your needs. Then, re-run the provisionning tool and you will end up with a bunch of folders:
```

Comment:
Explain what `ha` is. (high availability?)

Reply:
Done